gradle-idea-configuration/src/main/groovy/com/palantir/gradle/ideaconfiguration/PluginDependency.java

```diff
@@ -16,36 +16,43 @@
 
 package com.palantir.gradle.ideaconfiguration;
 
-import java.io.Serializable;
-import java.util.Objects;
 import java.util.regex.Pattern;
 import java.util.stream.IntStream;
+import org.gradle.api.Named;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
 
-public final class PluginDependency implements Serializable {
+public abstract class PluginDependency implements Named {
     private static final Pattern DOT_SPLITTER = Pattern.compile("\\.");
 
-    private final String name;
-    private String minVersion;
+    @Internal("Used to build up required min versions, not the final min version that is used")
+    public abstract SetProperty<String> getMinRequiredVersions();
 
-    public PluginDependency(String name) {
-        this.name = name;
-    }
+    @Input
+    @Override
+    public abstract String getName();
 
-    public String name() {
-        return name;
+    @Input
+    public final Provider<String> getMinVersion() {
+        return getMinRequiredVersions().map(requiredMinVersions -> {
```

Comment:
This approach has the added benefit that if someone wanted, they could supply a lazy min version that is the output of some other task, and gradle would automatically link up the task dependencies here.